src/legendhpges/icpc.py

```diff
@@ -0,0 +1,235 @@
+"""Build the inverted coaxial point contact crystal template."""
+import json
+import logging
+from math import pi, tan
+
+import numpy as np
+
+# Third-party imports
+import pyg4ometry as pg4
+
+
+class icpc:
+    """Define ICPC Germanium template."""
+
+    def __init__(self, jsonfile, reg, materials):
+        """
+        Create ICPC detector logical volume (LV).
+
+        Parameters
+        ----------
+        jsonfile : str
+            JSON input file name describing crystal shape.
+        reg : pg4.geant4.Registry, optional
+            if None, (almost, see next) standalone construction
+        materials : dict
+            predefined materials dictionary, required input.
+
+        Returns
+        -------
+        None.
+
+        """
+        # registry to store gdml data
+        # allow for independent construction
+        # BUT dependence on materials dictionary remains
+        if reg is None:
+            reg = pg4.geant4.Registry()
+
+        # read, then build the geometry
+        self.crystalLV = None
+        self.detname = ""
+
+        # add auxiliary info: type, value
+        aux = pg4.gdml.Defines.Auxiliary("SensDet", "GeDet", reg)
+
+        # build crystal, declare as detector
+        jsondict = self._read_from_file(jsonfile)
+        if jsondict is not None:
+            self.crystalLV = self._build_crystal(jsondict, reg, materials)
+            self.crystalLV.addAuxiliaryInfo(aux)
+
+    def __repr__(self):
+        """Print the object purpose."""
+        s = "ICPC detector: build a given ICPC from JSON input file.\n"
+        return s
+
+    def get_crystal_lv(self):
+        """
+        Access to logical volume as constructed here.
+
+        Returns
+        -------
+        pg4.geant4.LogicalVolume
+            Fully constructed LV of ICPC detector.
+
+        """
+        return self.crystalLV
+
+    def get_name(self):
+        """
+        Access to detector name from JSON file.
+
+        Returns
+        -------
+        name : str
+            Detector name string.
+
+        """
+        return self.detname
+
+    def _read_from_file(self, jsonfile):
+        """
+        Create crystal from description in JSON file.
+
+        Parameters
+        ----------
+        jsonfile : str
+            JSON input file name describing crystal shape.
+
+        Returns
+        -------
+        data : dict
+            Decoded JSON file content as dictionary.
+
+        """
+        try:
```

Comment:
I would remove the try-except block and let the program throw the exception - I would say this is the desired behaviour.